libs/api/domains/license-service/src/lib/licenseService.service.ts

```diff
@@ -353,21 +418,35 @@ export class LicenseServiceService {
       this.cacheManager,
     )
 
+    const metricsTags = this.getMetricsTags(licenseType)
+    this.metrics.increment(`pkpass_qrcode`, 1, metricsTags)
+
     if (licenseService) {
+      const start = DogStatsD.timer()
       pkpassQRCode = await licenseService.getPkPassQRCode(
```

Comment:
I'd suggest having a timing helper which wraps promises:

```ts
pkpassQRCode = await this.metrics.measure(
  'pkpass_qrcode.duration',
  metricsTags,
  licenseService.getPkPassQRCode(...)
)
```

Reply:
This reminds me that another fancy option is to track classes and function calls with APM:

https://github.com/DataDog/dd-trace-js/blob/master/docs/API.md#manual-instrumentation

This adds a span to the APM traces (and also supports all kinds of filtering and metrics). Main difference is that APM traces are expensive and you may lose data under load so they're not perfect for high-load metrics and probably not for this use case. But they can be very useful for debugging.